kairos-server/src/main.rs

```diff
@@ -1,21 +1,13 @@
-use std::net::SocketAddr;
-
 use dotenvy::dotenv;
-use kairos_server::{config::ServerConfig, state::BatchStateManager};
+use kairos_server::{config::ServerConfig, run};
 
 #[tokio::main]
 async fn main() {
-    tracing_subscriber::fmt::init();
```

Comment:
**Suggestion:** Initiate the subscriber as the first action in `main()`, as we may need to emit logs before the server is actually run.

Reply:
There is not a lot happening in `main` only sourcing the environment which is ok to fail, and the parsing of the config, which panics. I understand that we might want to add logs in the future here when the `env` sourcing fails, but If we move it to main again we loose the nice property that `run` does everything we need to run the server with everything it needs. Logging initialization would need to be done additionally. However, we could consider calling `init` twice. in main and `run`.

Reply:
If someone attempts to replace panics with logging in config parsing - an approach that seems quite likely - they MUST rework the subscriber initialization. However, it would be very easy to overlook this additional step and I would like to avoid creating this technical debt.

Where else is `run` called? If it's within `kairos-test-utils`, then I'm confident that we don't need logging there. Additionally, having the ability to configure subscribers externally seems like a desirable feature. I'm not very familiar with `tracing`; perhaps @Avi-D-coder could share his thoughts on this.

Reply:
The docs say that init panics if it's called twice: https://docs.rs/tracing-subscriber/latest/tracing_subscriber/fmt/fn.init.html

`run` ins called in `kairos-test-utils` thus basically anywhere where tests might be run. We ideally want our server to produce logs when a test fails or during development.

Reply:
I prefer `tracing_subscriber::fmt::init()` at the top of main, and I want it enabled in tests.

But it's a minor thing so I don't care until we need it.